markdown/src/org/intellij/plugins/markdown/structureView/MarkdownStructureViewFactory.java

```diff
@@ -38,10 +43,16 @@ protected boolean isSuitable(PsiElement element) {
         return false;
       }
       final PsiElement parent = element.getParent();
-      if (MarkdownStructureElement.hasTrivialChild(parent)) {
+      if (hasTrivialChild(parent)) {
```

Comment:
Is this acceptable after your change? I feel like it may ignore some elements incorrectly.